examples/particle-gibbs/script.jl

```diff
@@ -89,25 +91,25 @@ plot(x; label="x", xlabel="t")
 # 
 plot(y; label="y", xlabel="t")
 
-# Each model takes an `AbstractRNG` as input and generates the logpdf of the current transition:
-mutable struct NonLinearTimeSeries <: AdvancedPS.AbstractStateSpaceModel
-    X::Vector{Float64}
-    θ::Parameters
-    NonLinearTimeSeries(θ::Parameters) = new(Float64[], θ)
+# Define the model dynamics
+function SSMProblems.transition!!(rng::AbstractRNG, model::StochasticVolatilityModel)
+    return rand(rng, f₀(model))
 end
 
-# The dynamics of the model is defined through the `AbstractStateSpaceModel` interface:
-AdvancedPS.initialization(model::NonLinearTimeSeries) = f₀(model.θ)
-AdvancedPS.transition(model::NonLinearTimeSeries, state, step) = f(model.θ, state, step)
-function AdvancedPS.observation(model::NonLinearTimeSeries, state, step)
-    return logpdf(g(model.θ, state, step), y[step])
+function SSMProblems.transition!!(
+    rng::AbstractRNG, model::StochasticVolatilityModel, state::Float64, ::Int
+)
+    return rand(rng, f(state, model))
+end
+
+function SSMProblems.emission_logdensity(
+    model::StochasticVolatilityModel, state::Float64, observation::Float64, ::Int
+)
+    return logpdf(g(state, model), observation)
 end
-AdvancedPS.isdone(::NonLinearTimeSeries, step) = step > Tₘ
 
-# Here we use the particle gibbs kernel without adaptive resampling.
-model = NonLinearTimeSeries(θ₀)
 pg = AdvancedPS.PG(Nₚ, 1.0)
-chains = sample(rng, model, pg, Nₛ; progress=false);
+chains = sample(rng, model, pg, Nₛ; observations=y, progress=false);
```

Comment:
The main issue here is that `AbstractMCMC` does not forward data explicitly, it just assumes the obervables are internal to the model

Reply:
I see. One way around this would be to define something like

```
struct ConditionedSSM
    ssm::AbstractStateSpaceModel
    observations::Vector{Vector{Float64}}
end
```

though that's one more bit of API bloat.

Reply:
Doesn't have to be nested, or another field.
```julia
function emission_logensity(model::AbstractStateSpaceModel, step::Int) 
   return logpdf(Normal(), observations[step]) # Global scope
end
```
or 
```julia
function emission_logensity(model::AbstractStateSpaceModel, step::Int) 
   return logpdf(Normal(), model.observations[step]) 
end
```

It's a bit what `model() | (; y=observations)` would do in `DynamicPPL`